hash/sha256-hash.c

```diff
@@ -0,0 +1,91 @@
+/* sha256-hash.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#include <wolfssl/options.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/hash.h>
+
+#ifndef NO_SHA256
+void usage(void)
+{
+    printf("./run <file to hash>\n");
+    exit(-99);
+}
+#endif
+
+int main(int argc, char** argv)
+{
+    int ret = -1;
+#ifndef NO_SHA256
+    byte  hash[WC_SHA256_DIGEST_SIZE];
+    byte* rawInput;
+    FILE* inputStream;
+    char* fName = NULL;
+    int numBytes = 0;
+    int fileLength = 0;
+
+    if (argc < 2)
+        usage();
+    fName = argv[1];
+    printf("Hash input file %s\n", fName);
+
+    inputStream = fopen(fName, "rb");
+    if (inputStream == NULL) {
+        printf("ERROR: Unable to open file\n");
+        return -1;
+    }
+
+    /* find length of the file */
+    fseek(inputStream, 0, SEEK_END);
+    fileLength = (int) ftell(inputStream);
+    fseek(inputStream, 0, SEEK_SET); 
+
+    /* allocate buffer based on file size */
+    rawInput = (byte*) XMALLOC(fileLength*sizeof(byte), NULL,
```

Comment:
This example allocates one huge buffer for the entire file, which could be a problem for a large file. A better example would be to chunk in smaller sizes. I recently added an example of this in wolfCrypt test here:
https://github.com/wolfSSL/wolfssl/blob/master/wolfcrypt/test/test.c#L20420
For the SP non-blocking stuff.